core/templates/pages/topic-editor-page/modal-templates/rearrange-skills-in-subtopics-modal.template.html

```diff
@@ -6,52 +6,61 @@ <h3>
-               ng-class="{'skills-valid': !uncategorizedSkillSummaries.length, 'skills-invalid': uncategorizedSkillSummaries.length}"
-               dnd-list="uncategorizedSkillSummaries"
-               dnd-drop="onMoveSkillEnd(null)">
-        <strong>Uncategorized Skills</strong>
-        <div class="uncategorized-skills-container">
-          <md-card class="skill-id"
-                   ng-repeat="skillSummary in uncategorizedSkillSummaries track by $index"
-                   dnd-draggable="skillSummary"
-                   dnd-effect-allowed="move"
-                   dnd-dragstart="onMoveSkillStart(null, skillSummary)">
-            <div class="skill-card">
-              <a ng-if="!isSkillDeleted(skillSummary)"
-                 class="protractor-test-uncategorized-skill-card"
-                 ng-href="<[getSkillEditorUrl(skillSummary.getId())]>" target="_blank" rel="noopener">
-                 <[skillSummary.getDescription()]>
-              </a>
-            </div>
-          </md-card>
-          <p class="intro-card-message protractor-test-no-uncategorized-skills" ng-if="uncategorizedSkillSummaries.length === 0">
-            There are no unassigned skills left in the topic.
-          </p>
-        </div>
-      </md-card>
-      <md-card class="subtopic item-container protractor-test-subtopic-column"
-               ng-repeat="(subtopicIndex, subtopic) in subtopics"
-               dnd-list="subtopic.getSkillSummaries()"
-               dnd-drop="onMoveSkillEnd(subtopic.getId())">
-        <div class="subtopic-header">
-          <strong uib-tooltip="<[subtopic.getTitle()]>" tooltip-placement="bottom" ng-if="subtopic.getTitle().length < 25"><[subtopic.getTitle()]></strong>
-          <strong uib-tooltip="<[subtopic.getTitle()]>" tooltip-placement="bottom" ng-if="subtopic.getTitle().length >= 25"><[subtopic.getTitle().substring(0,15)]>...</strong>
-        </div>
-        <div class="subtopic-container">
-          <md-card class="skill-id"
-                   ng-repeat="skillSummary in subtopic.getSkillSummaries() track by $index"
-                   dnd-draggable="skillSummary"
-                   dnd-dragstart="onMoveSkillStart(subtopic.getId(), skillSummary)"
-                   dnd-effect-allowed="move">
-            <div class="skill-card">
-              <a ng-href="<[getSkillEditorUrl(skillSummary.getId())]>" class="protractor-test-subtopic-skill-description" target="_blank" rel="noopener">
-                 <[skillSummary.getDescription()]>
-              </a>
-            </div>
-          </md-card>
-        </div>
-      </md-card>
+      <div class="item-flex-container">
+        <md-card class="item-container uncategorized-skill-ids protractor-test-uncategorized-skills-container"
+                 ng-class="{'skills-valid': !$ctrl.uncategorizedSkillSummaries.length, 'skills-invalid': $ctrl.uncategorizedSkillSummaries.length}"
+                 dnd-list="uncategorizedSkillSummaries"
```

Comment:
Missing $ctrl? I'm curious.. did this not fail when you tested it?

Reply:
In general, please try to catch issues like this while testing or during a self-review of your code. I have seen this pattern in previous PRs too.

Reply:
@Showtim3 could you please make a list of the steps you took for testing? I think we would like to see a video.

Given that these issues are recurring, please provide this for every PR you submit from now onwards.

Reply:
Ah, sorry guys, I missed adding $ctrl here, but it is redundant here. It's needed inside nested ng-if's. This is a surely a problem but not a bug. Also this is being tested by an e2e test.

Reply:
Thanks @Showtim3, let's remove it if it is redundant.

Reply:
This is redundant for all the lines except for the lines 45-60, so for consistency, i have added that everywhere in this file.